### PR TITLE
Lightly refactor registration modal

### PR DESCRIPTION
**Pull request by whit537**
&larr; #5977 ([diff](https://github.com/whit537/osf.io/compare/debug-OSF-6447...refactor-registration-modal)) — #6050 &rarr;

## Purpose

I refactored the registration modal while debugging #5977. This PR contains some of the safer refactors.

## Changes

n/a

## Side effects

n/a

## Ticket

https://openscience.atlassian.net/browse/OSF-6751

## TODO

- [x] Fix Travis.
- [x] Rebase once #5977 lands.

### CODE REVIEW

**Comment by whit537** [comment]
Close/reopen to bump Travis.

**Comment by whit537** [comment]
After rebasing onto #5977 at a6efbb4739bc476ee8a01deeba37228aedfed1dc, it [appears](https://travis-ci.org/CenterForOpenScience/osf.io/jobs/149210199#L8830) that the timeValidation test is now heisenbuggy ([cf.](https://github.com/CenterForOpenScience/osf.io/pull/5977#issuecomment-236922483)). Hmm ...

**Comment by whit537** [comment]
Here's the log for the pikaday case:

```
LOG LOG: 'loc:', 'undefined'
LOG LOG: 'max:', '2020-08-01 13:17:45.62100'
LOG LOG: 'end:', '2020-08-03 13:17:45.61800'
LOG LOG: 'tod:', '2020-08-02 13:17:45.42500'
LOG LOG: 'e<m:', false
LOG LOG: 'e<t:', false
```

And for the timeValidation case:

```
LOG LOG: 'loc:', '2016-08-02 13:19:04.46700'
LOG LOG: 'max:', '2020-08-01 13:19:04.46700'
LOG LOG: 'end:', '2020-08-01 13:19:04.46700'
LOG LOG: 'tod:', '2020-08-02 13:19:04.31800'
LOG LOG: 'e<m:', true
LOG LOG: 'e<t:', true
```

Diff:

```diff
diff --git a/website/static/js/registrationModal.js b/website/static/js/registrationModal.js
index bb05d09..b43a8be 100644
--- a/website/static/js/registrationModal.js
+++ b/website/static/js/registrationModal.js
@@ -72,6 +72,13 @@ var RegistrationViewModel = function(confirm, prompts, validator) {
     self.embargoIsShortEnough = function (x, y, embargoLocalDateTime) {
         var max = getMaximumDate(embargoLocalDateTime),
             end = self.embargoEndDate();
+        console.log('loc:', embargoLocalDateTime === undefined ? 'undefined' : moment(embargoLocalDateTime).format('YYYY-MM-DD HH:mm:ss.SSSSS'));
+        console.log('max:', moment(max).format('YYYY-MM-DD HH:mm:ss.SSSSS'));
+        console.log('end:', moment(end).format('YYYY-MM-DD HH:mm:ss.SSSSS'));
+        console.log('tod:', moment(todayMaximum).format('YYYY-MM-DD HH:mm:ss.SSSSS'));
+        console.log('e<m:', end.isSameOrBefore(max));
+        console.log('e<t:', end.isSameOrBefore(todayMaximum));
+        console.log()
         return end.isSameOrBefore(max) && end.isSameOrBefore(todayMaximum);
     };
```

**Comment by whit537** [comment]
I believe this is due to the inversion of the condition, from `max.isBefore(end)` to `end.isSameOrAfter(max)`. I did that for readability, to make the logical form the same as the next condition, `end.isSameOrAfter(todayMaximum)`. What broke?

**Comment by whit537** [comment]
Right, so the code is confused here, on the edge case. For the `todayMaximum` check we accept a moment _exactly_ four years out as short enough, but strictly speaking the max check prior to refactoring considered a `maxEmbargoMoment` that was four years out to _not_ pass the test. In harmonizing the refactored condition with the `todayMaximum` condition, I broke the expected behavior. I think the best fix here is to preserve the discrepancy with an asymmetric condition; the `todayMaximum` side is on the chopping block in #6050 anyway.

**Comment by whit537** [comment]
What about the min side? We don't have a test for the edge case as we do with the max condition. I suppose that means the behavior is undefined. Let's change it to be consistent with the max side.

**Comment by whit537** [comment]
This is ready for review once #5977 lands.

**Comment by whit537** [comment]
Rebased on `develop` post-#5977. Previous head was d7a3c2a7932a6def31fd81246b1e304e36ee7065.
